tests/config_test.cc

```diff
@@ -0,0 +1,160 @@
+#include "cris/core/config/config.h"
+
+#include <gtest/gtest.h>
+
+#include <fmt/core.h>
+
+#include <chrono>
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+#include <thread>
+
+namespace cris::core {
+
+constexpr double kEps = 1e-6;
+
+namespace fs = std::filesystem;
+
+class ConfigTest : public testing::Test {
+   public:
+    ConfigTest()
+        : testing::Test()
+        , test_config_dir_(std::tmpnam(nullptr))
```

Comment:
Don't do this, you should minimize the gap between tmpnam and creation as much as possible, and check if it is really empty dir afterward.

Reply:
That is why we use create_directory without 's'. Then we don't need this check.

Reply:
create_directory is fine, but we still need the check, it's to ensure no one's currently touching the dir.

Reply:
Not applicable anymore.